### DISCUSSION TITLE
Add in items multiple images

### DISCUSSION THREAD

**Discussion by andybuki** [ksclarke/jiiify-presentation#118] [Q&A] [answered]
Hi i use your jiiify-presentation V3.0 to create manifests from my json files. 
I try to add all pages/images of a book to manifest.

`manifest.setCanvases(LIST_OF_CANVAS)); `
Unfortunately i add only the last ellement of the list. 
Can you advice me how should it look like?

My code looks:

           `String manifestID ="";
            String canvasID ="";
            String annoID ="";
            String imageID ="";
            String annoPageID ="";

            Canvas canvas = new Canvas("");
            AnnotationPage<PaintingAnnotation> annoPage = new AnnotationPage<>("");
            PaintingAnnotation anno = new PaintingAnnotation("",canvas);
            ImageContent imageContent = new ImageContent("");
            List<Canvas> canvases;

            for (int j = 0; j < pages.length(); j++) {
                JSONObject pagesObj = (JSONObject) pages.get(j);

                String pages_position="";
                String pages_image_file="";
                String pages_id="";
                String pages_document_id="";
                int number_of_pages = pages.length();

                if (pagesObj.has("pages_position"))
                    pages_position = (String) pagesObj.get("pages_position").toString();

                if (pagesObj.has("pages_image_file"))
                    pages_image_file = (String) pagesObj.get("pages_image_file").toString();

                if (pagesObj.has("pages_id"))
                    pages_id = (String) pagesObj.get("pages_id").toString();

                if (pagesObj.has("pages_document_id"))
                    pages_document_id = (String) pagesObj.get("pages_document_id").toString();

                JSONObject json = readJsonFromUrl("https://iiif-content.crossasia.org/xasia/dllm"+"+dllm_000"+pages_document_id+"+"+pages_id+"/info.json");

                manifestID = SERVER + MANIFEST_COLLECTION + "000" + pages_document_id +"+"+ pages_id;
                canvasID = manifestID +"/canvas";
                imageID = manifestID+"/full/full/0/default.jpg";
                annoID = manifestID+"/annotation";
                annoPageID = manifestID;

                canvas = new Canvas(canvasID).setWidthHeight((Integer) json.get("width"), (Integer) json.get("height"));

                imageContent = new ImageContent(imageID).setWidthHeight((Integer) json.get("width"), (Integer) json.get("height"));
                annoPage = new AnnotationPage<>(annoPageID);
                anno = new PaintingAnnotation(annoID, canvas);
                annoPage.addAnnotations(anno.setBody(imageContent).setTarget(canvasID));

                manifest.setCanvases(canvas.setPaintingPages(annoPage));
            }`

Best regards
Andrey

**Comment by andybuki** [comment, marked as answer]
Hi, i allready found, how to solve it. Sometimes just to do it next day.

**Reply by ksclarke** [reply]
Glad you found the solution already. That's always the way it goes, isn't it? Put it down for a bit and the answer comes :-)

**Reply by ksclarke** [reply]
I should probably leave a little more explanation for the next person that comes along and reads this. The `setCanvases()` method sets all the canvases at once, so if you put it into a loop like the above only the last canvas is actually getting set.

The way I'd do it is to create a list of canvases outside the loop and add to that list in the loop, finally using `setCanvases(list)` once the loop is done with its work; or, alternatively, you could do something like `manifest.getCanvases().add(canvas)` inside the loop, which would add one canvas to the manifest's internal list of canvases on each iteration.
